rpmio/macro.c

```diff
@@ -1197,17 +1214,106 @@ static void doTrace(MacroBuf mb, int chkexist, int negate,
 static const char *setNegateAndCheck(const char *str, int *pnegate, int *pchkexist) {
 
     *pnegate = 0;
-    *pchkexist = 0;
+    *pchkexist = CHK_NO;
     while ((*str == '?') || (*str == '!')) {
 	if (*str == '!')
 	    *pnegate = !*pnegate;
 	else
-	    (*pchkexist)++;
+	    *pchkexist = CHK_BASIC;
 	str++;
     }
     return str;
 }
 
+static void partsInit(macroPartition *parts)
+{
+    parts->f = parts->fe = NULL;
+    parts->g = parts->ge = NULL;
+    parts->h = parts->he = NULL;
+    parts->fn = parts->gn = parts->hn = 0;
+    return;
+}
+
+static void setPartsSize(macroPartition *parts)
+{
+    parts->fn = parts->fe - parts->f;
+    parts->gn = parts->ge - parts->g;
+    parts->hn = parts->he - parts->h;
+    return;
+}
+
+/**
+ * Skip spaces before the input string
+ * @param str		string that starts after the ..
+ * @
+ */
```

Comment:
In general, doxygen annotations don't belong to internal helper functions, trivial in particular.